planning/route_handler/src/route_handler.cpp

```diff
@@ -587,12 +587,13 @@ lanelet::ConstLanelets RouteHandler::getLaneletSequenceUpTo(
       }
       candidate_lanelets = prev_lanes;
     }
-    // loop check
-    if (std::any_of(
-          candidate_lanelets.begin(), candidate_lanelets.end(),
-          [lanelet](auto & prev_llt) { return lanelet.id() == prev_llt.id(); })) {
-      break;
-    }
+
+    // loop check, remove goal lanelet from candidate lanelets
+    candidate_lanelets.erase(
+      std::remove_if(
+        candidate_lanelets.begin(), candidate_lanelets.end(),
+        [lanelet](auto & prev_llt) { return lanelet.id() == prev_llt.id(); }),
+      candidate_lanelets.end());
```

Comment:
Thanks for the fix.
Can you check if the mission planner will not die in some corner cases like putting the goal pose behind the ego in the ego's lanelet.

Reply:
I've verified different cases and didn't observe that mission planner die. Putting the goal behind the ego produce bad trajectory but this is known issue i supposed: 
![image](https://github.com/autowarefoundation/autoware.universe/assets/2842422/6dcbc036-fb5c-4313-99aa-30d6839dc125)

Reply:
Thank you. I will approve the change!